Data hub accessiblity fixes

sagar-lcm opened the pull request:
## Description
The following ticket's changes are present in the PR:

- https://massgov.atlassian.net/browse/DP-26730
- https://massgov.atlassian.net/browse/DP-26734
- https://massgov.atlassian.net/browse/DP-26747
- https://massgov.atlassian.net/browse/DP-26746
- https://massgov.atlassian.net/browse/DP-26749

sag13684 committed:
DP-26730: Add autocomplete attribute to HeaderSearch molecule

sag13684 committed:
DP-26730: Add autocomplete attribute to HeaderSearch molecule

sag13684 committed:
DP-26730: Add autocomplete attribute to HeaderSearch molecule

sag13684 committed:
DP-26747: Added disabled property for SelectBox

sag13684 committed:
DP-26734: Added visually hidden form title to SearchBannerForm

sag13684 committed:
DP-26734: Added visually hidden form title to SearchBannerForm

sag13684 committed:
DP-26734: Added visually hidden form title to SearchBannerForm

sag13684 committed:
DP-26734: Added visually hidden form title to SearchBannerForm

sag13684 committed:
DP-26734: Added visually hidden form title to SearchBannerForm

sag13684 committed:
DP-26734: Added visually hidden form title to SearchBannerForm

sag13684 committed:
DP-26746: Added clear all button context for Tags molecule

sag13684 committed:
DP-26746: Added clear all button context for Tags molecule

sag13684 committed:
DP-26746: Added clear all button context for Tags molecule

sag13684 committed:
DP-26749: After clicking a new page, allow screen reader users to go back to top

sag13684 committed:
DP-26749: After clicking a new page, allow screen reader users to go back to top

sag13684 committed:
DP-26743: Added role attribute to InputCheckbox

sag13684 committed:
Merge branch 'develop' of github.com:massgov/mayflower into data_hub_accessiblity_fixes

sag13684 committed:
DP-26743: Added role attribute to InputCheckbox

sag13684 committed:
DP-26743: Added role attribute to InputCheckbox

sag13684 committed:
DP-26752: Allows extra properties to be passed to BreadcrumbItem

sag13684 committed:
DP-26749: Added class to back to top pagination link

clairesunstudio committed:
Merge branch 'develop' into data_hub_accessiblity_fixes

clairesunstudio committed:
linting

ygannett reviewed (approved):
LGTM.  Thank you for all a11y improvements.

ygannett commented:
@sag13684 Can you look into the 2 failures?  They seem to be related.  As the CSS gets fixed, the backstop error should go away.  Once it's fixed, the PR can be merged.  Thank you!

https://output.circle-artifacts.com/output/job/3444f156-a874-4fd3-9c63-f1aa0a5b39e0/artifacts/0/backstop/data/html_report/index.html

sag13684 committed:
DP-26749: Adds styles to back to top pagination link

sag13684 committed:
Merge branch 'data_hub_accessiblity_fixes' of github.com:massgov/mayflower into data_hub_accessiblity_fixes

sag13684 committed:
Style lint fixes

sag13684 committed:
DP-26749: Adds styles to back to top pagination link

sag13684 committed:
DP-26747: Added disabled property for SelectBox

clairesunstudio commented on changelogs/DP-26730.yml:
```diff
@@ -0,0 +1,58 @@
+#___EXAMPLE___
+Added:
+  - project: React
+    description: Add autocomplete attribute to HeaderSearch molecule.
```
```suggestion
    description: Add autocomplete attribute to HeaderSearch molecule. (#1715)
```
can you please add the PR number to the changelogs, like this?

sagar-lcm replied:
Done.

clairesunstudio commented:
Something seems to broken the two-column styles of the LinkList component in backstop --> https://output.circle-artifacts.com/output/job/3f2c9368-0646-49f8-8ca2-a53172e4e807/artifacts/0/backstop/data/html_report/index.html

sag13684 committed:
Added PR numbers to the change log description

clairesunstudio commented:
You can ignore the backstop test failure on LinkList, it's introduced by this PR https://github.com/massgov/mayflower/pull/1705

clairesunstudio commented on changelogs/DP-26734.yml:
```diff
@@ -0,0 +1,58 @@
+#___EXAMPLE___
+Added:
+  - project: React
+    description: Added visually hidden form title to SearchBannerForm.
```
```suggestion
    description: Added visually hidden form title to SearchBannerForm. (#1715)
```

sagar-lcm replied:
Done.

clairesunstudio commented on changelogs/DP-26743.yml:
```diff
@@ -0,0 +1,58 @@
+#___EXAMPLE___
+Added:
+  - project: React
+    description: Added role attribute to InputCheckbox.
```
```suggestion
    description: Added role attribute to InputCheckbox. (#1715)
```

clairesunstudio commented on changelogs/DP-26746.yml:
```diff
@@ -0,0 +1,58 @@
+#___EXAMPLE___
+Added:
+  - project: React
+    description: Added clear all button context for Tags molecule.
```
```suggestion
    description: Added clear all button context for Tags molecule. (#1715)
```

clairesunstudio commented on changelogs/DP-26747.yml:
```diff
@@ -0,0 +1,58 @@
+#___EXAMPLE___
+Added:
+  - project: React
+    description: Added disabled property for SelectBox.
```
```suggestion
    description: Added disabled property for SelectBox. (#1715)
```

clairesunstudio commented on changelogs/DP-26749.yml:
```diff
@@ -0,0 +1,58 @@
+#___EXAMPLE___
+Added:
+  - project: React
+    description: Added back to top property for Pagination molecule.
```
```suggestion
    description: Added back to top property for Pagination molecule. (#1715)
```

clairesunstudio commented on changelogs/DP-26752.yml:
```diff
@@ -0,0 +1,58 @@
+#___EXAMPLE___
+Added:
+  - project: React
+    description: Allows extra properties to be passed to BreadcrumbItem.
```
```suggestion
    description: Allows extra properties to be passed to BreadcrumbItem. (#1715)
```

sag13684 committed:
DP-26222: Fix failing backstop test for link list

clairesunstudio committed:
Revert "DP-26222: Fix failing backstop test for link list"

This reverts commit 8a7ad282e0aba4669d334b4b59cfbd8b40a7bfe7.

clairesunstudio commented:
The test failure is not related to this PR. I created a separated ticket to address that on develop

clairesunstudio committed:
Merge branch 'develop' into data_hub_accessiblity_fixes

clairesunstudio commented:
patternlab backstop test failure is a false positive

clairesunstudio committed:
Merge branch 'develop' into data_hub_accessiblity_fixes

DmitriiTartynovLCM committed:
DP-26714: Refactor GenTeaser into multiple separate components (#1717)

* DP-26714: Split GenTeaser component into a set of independent components.

Maintain backward compatibility, so that we don't have to release another major release.

* DP-26714: Update the docs & the Storybook story with the GenTeaser refactoring.

* DP-26714: Fix code style issues.

* DP-26714: Add changelog.

* DP-26714: Remove internal components - TeaserOrgs and TeaserSearch. Update the changelog.

* DP-26714: Document the legacy usage method of GenTeaser.

Co-authored-by: Minghua Sun <clairesunstudio@gmail.com>

DmitriiTartynovLCM committed:
DP-26745: A11y improvements to `Tags` and `ButtonTag` (#1716)

* DP-26745: Let the button tag content be a node, not just plain text.

* DP-26745: Add changelog.

* DP-26745: Render the icon with HTML rather than CSS. Add description prop.

* Hide the icon from screen readers.

* style lint

Co-authored-by: Minghua Sun <clairesunstudio@gmail.com>

clairesunstudio committed:
Merge branch 'develop' into data_hub_accessiblity_fixes
